### PR TITLE
Aws route53 fixes

### PR DESCRIPTION
**Pull request by rvanderwerf**
I ran through all of the integration tests and they seem to be working. I found some issues with the camel casing of properties that was breaking things, also properties created with Boolean flags on test startup had to be escaped with single quotes i.e. 'true' instead of the the direct value.

### CODE REVIEW

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                if (result.getOperation().getStatus().equalsIgnoreCase("failure")) {
+                    if (route53AutoRegistrationConfiguration.isFailFast()) {
+                        LOG.error("Error registering instance shutting down instance.");
+                        //TODO how do I get the current instance?
+                        ((EmbeddedServerInstance) instance).getEmbeddedServer().stop();
+                    }
+                }
+            }
+        }
+        LOG.info("Simple Job every 10 seconds :{}", new SimpleDateFormat("dd/M/yyyy hh:mm:ss").format(new Date()));
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: please wrap with `if (LOG.isInfoEnabled())` also the log message is a bit generic, is it needed? The message doesn't say it is related to Route53 stuff...

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            String operationId = buf.readLine();
+            GetOperationRequest operationRequest = new GetOperationRequest().withOperationId(operationId);
+            GetOperationResult result = route53AutoNamingRegistrationClient.getDiscoveryClient().getOperation(operationRequest);
+            if (result.getOperation().getStatus().equalsIgnoreCase("failure") || result.getOperation().getStatus().equalsIgnoreCase("success")) {
+                LOG.info("Service regisration for operation "+operationId+" resulted in "+result.getOperation().getStatus());
+                fileInputStream.close();
+                operationFile.delete();
+                if (result.getOperation().getStatus().equalsIgnoreCase("failure")) {
+                    if (route53AutoRegistrationConfiguration.isFailFast()) {
+                        LOG.error("Error registering instance shutting down instance.");
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: please wrap with `if (LOG.isErrorEnabled())`

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        // log for status file with an active operation id
+        File operationFile = new File(operationStatusFile);
+        if (operationFile.exists()) {
+            FileInputStream fileInputStream = new FileInputStream(operationFile);
+            BufferedReader buf = new BufferedReader(new InputStreamReader(fileInputStream));
+            String operationId = buf.readLine();
+            GetOperationRequest operationRequest = new GetOperationRequest().withOperationId(operationId);
+            GetOperationResult result = route53AutoNamingRegistrationClient.getDiscoveryClient().getOperation(operationRequest);
+            if (result.getOperation().getStatus().equalsIgnoreCase("failure") || result.getOperation().getStatus().equalsIgnoreCase("success")) {
+                LOG.info("Service regisration for operation "+operationId+" resulted in "+result.getOperation().getStatus());
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: please wrap with `if (LOG.isInfoEnabled())`

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(ServiceRegistrationStatusJob.class);
+
+    @Value("${server.serviceRegisterFile:/tmp/serviceOperationStatus.txt}")
+    String operationStatusFile;
+
+    @Inject
+    Route53AutoNamingRegistrationClient route53AutoNamingRegistrationClient;
+
+    @Inject
+    Route53AutoRegistrationConfiguration route53AutoRegistrationConfiguration;
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: Could you change this to use constructor injection instead. Field injection is not really recommended.

File: cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy [hunk 5]
```diff
@@ -133,5 +133,7 @@ class DefaultFeature implements Feature {
         if (maxJavaVersion != null) {
             return javaVersion <= maxJavaVersion
         }
+
+        true
```
Comment (graemerocher) [inline cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy]: Is this change necessary?

File: cli/src/main/groovy/io/micronaut/cli/profile/repository/MavenProfileRepository.groovy [hunk 6]
```diff
@@ -139,6 +140,26 @@ class MavenProfileRepository extends AbstractJarProfileRepository {
...
+            }
+
+            if(!mavenLocalLocation) {
+                mavenLocalLocation = "${System.getProperty("user.home")}/.m2/repository/"
+            }
+        }
+
+        mavenLocalLocation
+    }
+
```
Comment (graemerocher) [inline cli/src/main/groovy/io/micronaut/cli/profile/repository/MavenProfileRepository.groovy]: Could you separate out the CLI changes into a separate pull request? They seem unrelated to the Route53 changes.

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/client/Route53AutoNamingClient.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    @Inject
+    AWSClientConfiguration awsClientConfiguration;
+
+    @Inject
+    Route53ClientDiscoveryConfiguration route53ClientDiscoveryConfiguration;
+
+    @Inject
+    Route53DiscoveryConfiguration route53DiscoveryConfiguration;
+
+    AWSServiceDiscoveryAsync discoveryClient;
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/client/Route53AutoNamingClient.java]: Please change to use constructor injection

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/client/Route53AutoNamingClient.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    private void init() {
+        discoveryClient = AWSServiceDiscoveryAsyncClientBuilder.standard().withClientConfiguration(awsClientConfiguration.getClientConfiguration()).build();
+    }
+
+    /*@Override
+    public Publisher<List<ServiceInstance>> getInstances(String serviceId) {
+        Publisher<ListInstancesResult> instancesResultPublisher = getAwsInstances(serviceId);
+        instancesResultPublisher.subscribe();
+
+    }*/
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/client/Route53AutoNamingClient.java]: Maybe delete this if it is not necessary

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/Route53AutoNamingRegistrationClient.java [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+        this.discoveryClient = discoveryClient;
+    }
+
+    public Service getDiscoveryService() {
+        return discoveryService;
+    }
+
+    public void setDiscoveryService(Service discoveryService) {
+        this.discoveryService = discoveryService;
+    }
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/Route53AutoNamingRegistrationClient.java]: Please add missing java doc for methods above

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
+package io.micronaut.discovery.aws.route53.registration;
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: Please add missing license header

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@Singleton
```
Comment (graemerocher) [inline discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: Please add missing javadoc

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy [hunk 12]
```diff
@@ -976,9 +935,6 @@ class InjectTransform implements ASTTransformation, CompilationUnitAware {
                 return
             }
             AnnotationMetadata fieldAnnotationMetadata = AstAnnotationUtils.getAnnotationMetadata(fieldNode)
-            typeElementVisitors.findAll { it.matches(fieldAnnotationMetadata) }.each {
-                it.visit(fieldNode, fieldAnnotationMetadata)
-            }
```
Comment (graemerocher) [inline inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy]: There seem to be multiple changes to `InjectTransform` that are unrelated. Please rebase and remove these changes

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/TypeElementVisitorTransform.groovy [hunk 13]
```diff
@@ -0,0 +1,183 @@
+/*
```
Comment (graemerocher) [inline inject-groovy/src/main/groovy/io/micronaut/ast/groovy/TypeElementVisitorTransform.groovy]: These changes and everything below seem unrelated?  Please rebase.

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy (hunk 12 above)
Comment (rvanderwerf) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy]: I must have picked these up from a merge from master? I am not sure.

File: discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java (hunk 1 above)
Comment (rvanderwerf) [reply discovery-client/src/main/java/io/micronaut/discovery/aws/route53/registration/ServiceRegistrationStatusJob.java]: I will move this to a runnable so it's not running forever after it starts like you suggested in the meeting.

File: cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy (hunk 5 above)
Comment (rvanderwerf) [reply cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy]: The only place I can think of it came in from a merge from master?

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy (hunk 12 above)
Comment (rvanderwerf) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy]: weird this case shows as matching up with master 100%

File: cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy (hunk 5 above)
Comment (rvanderwerf) [reply cli/src/main/groovy/io/micronaut/cli/profile/DefaultFeature.groovy]: when I compare this file to origin/master it says 100% match

File: cli/src/main/groovy/io/micronaut/cli/profile/repository/MavenProfileRepository.groovy (hunk 6 above)
Comment (rvanderwerf) [reply cli/src/main/groovy/io/micronaut/cli/profile/repository/MavenProfileRepository.groovy]: hmm there shouldn't be any changes here.. when I compare this file with master it says 100% identical.

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy (hunk 12 above)
Comment (rvanderwerf) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy]: hmm I'm having some issues rebasing.. I takes me through hundreds of other commits then dies :(

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/TypeElementVisitorTransform.groovy (hunk 13 above)
Comment (rvanderwerf) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/TypeElementVisitorTransform.groovy]: yes this is super confusing when I compare with origin/master it says 100% identical. I'm having issues doing a successful rebase though, I think IntelliJ barfs part way through :(

**Comment by graemerocher** [comment]
@rvanderwerf I have tried to cleanup the code a bit. There is however one failing test against master.

Can you verify my changes, test that everything works and resolve the failing test? Then I think it will be ready to merge
